ocean_lib/web3_internal/contract_base.py

```diff
@@ -227,6 +228,7 @@ def send_transaction(
             "from": from_wallet.address,
             "passphrase": from_wallet.password,
             "account_key": from_wallet.key,
+            "chainId": contract_function._contract_function.web3.eth.chain_id
```

Comment:
I think it's better to pull the Web3 instance from the `ContractBase.w3` instance variable instead of the `CustomContractFunction` you've created here.

Please change to 

```python
"chainId": self.w3.eth.chain_id,
```